include/nlohmann/detail/exceptions.hpp

```diff
@@ -141,20 +141,24 @@ class exception : public std::exception
   private:
     /// an exception object as storage for error messages
     std::runtime_error m;
+#if JSON_DIAGNOSTIC_POSITIONS
     template<typename BasicJsonType>
     static std::string get_byte_positions(const BasicJsonType* leaf_element)
     {
-#if JSON_DIAGNOSTIC_POSITIONS
         if ((leaf_element->start_pos() != std::string::npos) && (leaf_element->end_pos() != std::string::npos))
         {
-            return concat('(', "bytes ", std::to_string(leaf_element->start_pos()), "-", std::to_string(leaf_element->end_pos()), ") ");
+            return concat("(bytes ", std::to_string(leaf_element->start_pos()), "-", std::to_string(leaf_element->end_pos()), ") ");
         }
         return "";
+    }
 #else
+    template<typename BasicJsonType>
+    static std::string get_byte_positions(const BasicJsonType* leaf_element)
```

Comment:
@nlohmann Do you have a preference on providing the parameter name and using `static_cast` to mark it as used, or leaving the parameter name off to say it's always unused?

Since this isn't isn't being used for documentation, my preference would be to leave the name off.

Reply:
Yes, leaving the name off should be fine.

Reply:
> error: all parameters should be named in a function [hicpp-named-parameter,readability-named-parameter,-warnings-as-errors]

Okay, who thought THAT was a good clang tidy check???

>The check implements the following rule originating in the Google C++ Style Guide:

Oh, well that explains it.  I find myself disagreeing with their style guide on a regular basis.